mmdet/core/bbox/iou_calculators/iou2d_calculator.py

```diff
@@ -32,17 +51,99 @@ def __call__(self, bboxes1, bboxes2, mode='iou', is_aligned=False):
             bboxes2 = bboxes2[..., :4]
         if bboxes1.size(-1) == 5:
             bboxes1 = bboxes1[..., :4]
+
+        if self.dtype == 'fp16':
+            # change tensor type to save cpu and cuda memory and keep speed
+            bboxes1 = cast_tensor_type(bboxes1, self.scale, self.dtype)
```

Comment:
Can we move the logic of lines 55-63 into `bbox_overlaps`? This makes `bbox_overlaps` a more general function that enables FP16 and simplifies the logic in BboxOverlaps2D.__call__.

Reply:
enen, thank you for your advice.

Reply:
yeah, of course you can.

Reply:
OK  agree